utils/symbol.c

```diff
@@ -1430,3 +1423,71 @@ void print_symtabs(struct symtabs *symtabs)
 		symbol_putname(sym, name);
 	}
 }
+
+static unsigned long kaddr;
+
+static unsigned long get_kaddr(char *str)
+{
+	char *pch;
+	unsigned long addr;
+
+	pch = strtok (str,"-");
+	addr = strtoul(pch, NULL, 16);
+
+	if (addr < 0x40000000UL) {
+		return 0x40000000UL;
+	} else if (addr < 0x80000000UL) {
+		return 0x80000000UL;
+	} else if (addr < 0xB0000000UL) {
+		return 0xB0000000UL;
+	} else if (addr < 0xC0000000UL) {
+		return 0xC0000000UL;
+#if __SIZEOF_LONG__ == 8
+	} else {
+		return 0x800000000000ULL;
+	}
+#else
+    }
+#endif
```

Comment:
Please do not use the `#if` directives for a different block boundary.  In fact, this function is meaninful only for 32-bit (ARM) systems so that you can omit the 64-bit part entirely IMHO.